x-pack/plugins/monitoring/server/plugin.ts

```diff
@@ -325,8 +325,22 @@ export class Plugin {
             getKibanaStatsCollector: () => this.legacyShimDependencies.kibanaStatsCollector,
             getUiSettingsService: () => context.core.uiSettings.client,
             getActionTypeRegistry: () => context.actions?.listTypes(),
-            getAlertsClient: () => plugins.alerts.getAlertsClientWithRequest(req),
-            getActionsClient: () => plugins.actions.getActionsClientWithRequest(req),
+            getAlertsClient: () => {
+              try {
+                return plugins.alerts.getAlertsClientWithRequest(req);
+              } catch (err) {
+                // If security is disabled, this call will throw an error unless a certain config is set for dist builds
+                return null;
+              }
+            },
+            getActionsClient: () => {
+              try {
+                return plugins.actions.getActionsClientWithRequest(req);
+              } catch (err) {
+                // If security is disabled, this call will throw an error unless a certain config is set for dist builds
+                return null;
+              }
+            },
```

Comment:
nit: This works, but I'm wondering if we might also be swallowing other "legitimate" errors. Maybe we can do a check instead? eg:
```javascript
getAlertsClient: () => {
   return setupPlugins.encryptedSavedObjects.usingEphemeralEncryptionKey ? null : plugins.actions.getActionsClientWithRequest(req);
}
```

Reply:
I think we can do that as an enhancement. I'm worried about looking at something other than the alerting plugin to figure out if alerting will work. If they remove that at some point, we would have to know to remove ours too.